Lib/Allocator.cpp

```diff
@@ -33,18 +33,27 @@ void Lib::setMemoryLimit(size_t limit) { LIMIT = limit; }
 
 // override global allocators to keep track of allocated memory, doing very little else
 // TODO does not support get_new_handler/set_new_handler as we don't use it, but we could
-void *operator new(size_t size) {
+void *operator new(size_t size, std::align_val_t align) {
   if(ALLOCATED + size > LIMIT)
     throw std::bad_alloc();
   ALLOCATED += size;
   {
     Lib::TimeoutProtector tp;
-    if(void *ptr = std::malloc(size))
+    if(void *ptr = std::aligned_alloc(static_cast<size_t>(align), size))
       return ptr;
   }
   throw std::bad_alloc();
 }
 
+// align-less operator new
+// forwards to the aligned version with alignof(max_align_t)
+void *operator new(size_t size) {
+  return operator new(
+    size,
+    static_cast<std::align_val_t>(alignof(std::max_align_t))
```

Comment:
Maybe this is why we get so many `align=16` requests to the aligned version?

Reply:
Because we say we want everything aligned "maximally"? (I thought you wrote this code, @MichaelRawson ;))